packages/blocks/src/image-block/image-block.ts

```diff
@@ -411,6 +413,37 @@ export class ImageBlockComponent extends BlockElement<ImageBlockModel> {
     };
   }
 
+  private _onCaptionKeydown(e: KeyboardEvent) {
+    if (e.key === 'Enter') {
+      e.stopPropagation();
+      const model = this.model;
+      const page = model.page;
+      const target = e.target as HTMLInputElement;
+      const start = target.selectionStart;
+      if (start === null) return;
+
+      const value = target.value;
+      const caption = (this._caption = value.slice(0, start));
+      target.value = caption;
+      page.updateBlock(model, { caption });
+
+      const nextBlockText = value.slice(start);
+      const parent = page.getParent(model);
+      if (!parent) return;
+      const index = parent.children.indexOf(model);
+      const id = page.addBlock(
+        'affine:paragraph',
+        { text: new Text(nextBlockText) },
+        parent,
+        index + 1
+      );
+      asyncFocusRichText(model.page, id, {
+        index: nextBlockText.length,
```

Comment:
I think it will be more reasonable to use `0` to keep index when we split the text.